src/main/java/serverutils/handlers/ServerUtilitiesWorldEventHandler.java

```diff
@@ -95,4 +96,29 @@ public void onExplosionDetonate(ExplosionEvent.Detonate event) {
             }
         }
     }
+
+    @SubscribeEvent
+    public void onWorldLoad(WorldEvent.Load event) {
+        if (!event.world.isRemote) {
+            int dimensionId = event.world.provider.dimensionId;
+            MinecraftServer server = MinecraftServer.getServer();
+            if (!ChunkLoaderManager.instance.isGenerating()
+                    && !ChunkLoaderManager.instance.initializeFromPregeneratorFiles(server, dimensionId)) {
+                ChunkLoaderManager.instance.reset(false);
+                System.out.println("No pregenerator to load for dimension Id: " + dimensionId);
```

Comment:
This should not log unconditionally on every world load.
All `System.out.println` should also be replaced by a proper logger, you can either use `ServerUtilities.LOGGER` or create a new logger to be used for the pregen.

Reply:
Whoops. That was a mistake left over from me porting it over after just doing rough testing. Will change this.